.github/workflows/build-and-push-to-ghcr.yml

```diff
@@ -0,0 +1,41 @@
+name: Build and push to GHCR
+on:
+  push:
+    branches: [main]
+  pull_request:
+env:
+  GHCR_URL: ghcr.io/coqui-ai/xtts-streaming-server
+jobs:
+  build-and-push-to-ghcr:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set Release version
+        shell: bash
+        run: |
+          if [[ "$GITHUB_EVENT_NAME" == "pull_request" ]]; then
+            echo "RELEASE_VERSION=dev" >> $GITHUB_ENV
+          else
+            echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+          fi 
+          export TAG=${GHCR_URL}:${{ env.RELEASE_VERSION }}-${{ github.sha }}
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: 'Login to GitHub Container Registry'
+        run: |
+          set -xe
+          docker login --username ${{ github.actor }} --password ${{ secrets.GITHUB_TOKEN }} ghcr.io
+
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          file: server/Dockerfile
+          push: false
```

Comment:
you need `${{ }}` to embed logic. E.g. `${{ env.RELEASE_VERSION == "main" }}`

Docs here: https://docs.github.com/en/actions/learn-github-actions/expressions

Reply:
tried on push : false, that does not work as it is considered yaml, added 2 steps with checks on main

Reply:
you tried with incorrect syntax